pkg/utils/sliceutils.go

```diff
@@ -399,3 +399,11 @@ func SelectFromSlice[T any](slice []T, shouldKeep func(T) bool) []T {
 
 	return result
 }
+
+func NormalizeSlice[T any](slice []T) []T {
+	if slice == nil {
+		return []T{}
+	}
+
+	return slice
+}
```

Comment:
Just curious: in what situations is having an empty slice instead of a nil slice beneficial? Go has some nice properties with nil maps, but I'm not sure about nil slices

Reply:
So I had to do this because, when we do `reflect.DeepEqual`, this is causing an issue when there is a `nil` slice and you are comparing with slice empty slice.

So I created this for that purpose.